Fixed: Parsing of subtitle languages separated by dash

markus101 opened the pull request:
#### Description

Allow for additional separators other than period between ISO language and additional subtitle information.


#### Issues Fixed or Closed by this PR
* Closes #6494

JeWe37 commented:
I believe this may cause problems. Subtitle titles with spaces are fairly common, and I believe e.g.
```
Name (2020) - S01E20 - [AAC 2.0].Full Subtitles.eng.ass
```
will break this. I would argue that in order to accept a title, it should be in the narrowly defined format from before, and rely on the solution proposed in [my previous comment](https://github.com/Sonarr/Sonarr/issues/6494#issuecomment-1947769387) to ignore the title.

JeWe37 commented:
I think you're actually right about the EDIT, this came to mind later and I forgot to account for the fact that this still requires a known tag or any alpha string of less 2 or 3 characters to be present in the title preceding the number.

The "parse as full and clean up the subtitle tag" admittedly doesn't sound great to me. Which makes me ask, does anything actually support the full flag? 
- Plex doesn't based on its docs https://support.plex.tv/articles/200471133-adding-local-subtitles-to-your-media/
- Jellyfin/Emby definitely doesn't https://github.com/jellyfin/jellyfin/blob/ef96b5fa20f6f09e2671cf20efd075f134274314/Emby.Naming/Common/NamingOptions.cs#L291-L308
- Kodi doesn't seem to support anything except forced https://kodi.wiki/view/Subtitles

I think just removing this flag should actually cover most of the breakage based on a brief check in my own library. All the subs I saw where you had another tag that would get parsed incorrectly always just had that tag as the title. I.e. something like `Forced` as the title rather than `Forced Subtitles` so that reinterpretation may actually be valid.

markus101 commented:
`full` does seems a little weird, not even sure what it'd mean other than the full episode, but then Sonarr wouldn't really care because a partial file isn't really supported. If the file has that it'd be considered part of the title and extracted, but that's reasonable.

I've added a commit that drops full and adds some test cases around the things mentioned here and of course the original issue is addressed.
